tests/core/cluster_common.py

```diff
@@ -0,0 +1,96 @@
+import random
+import subprocess
+import os
+import re
+import yaml
+from jinja2 import Template
+
+
+def generate_cluster_config(dind_rke_node_num):
+    # generate a random and kube_config file
+    dind_random = random.randint(1, 100000)
+    dind_name = 'dind-' + str(dind_random)
+    dind_cluster_config_file = dind_name + '.yml'
+    dind_kube_config_file = 'kube_config_' + dind_name + '.yml'
+    cluster_config_tmpl = get_rke_config_template()
+    # generate nodes
+    random_nodes = [
+            'node-' +
+            str(random.randint(10000, 99999))
+            for x in range(dind_rke_node_num)]
+    rke_config_template = Template(cluster_config_tmpl)
+    rendered_tmpl = rke_config_template.render(
+        random_nodes=random_nodes)
+    # write config file on disk
+    cluster_config_file = open(dind_cluster_config_file, 'w')
+    cluster_config_file.write(rendered_tmpl)
+    cluster_config_file.close()
+    return \
+        dind_name, \
+        yaml.safe_load(rendered_tmpl), \
+        dind_cluster_config_file, \
+        dind_kube_config_file
+
+
+def get_rke_config_template():
+    dind_cluster_config_j2 = """
+---
+authentication:
+    strategy: "x509|webhook"
+nodes:{% for node in random_nodes %}
+  - address: {{ node }}
+    user: docker
+    role:
+    - controlplane
+    - worker
+    - etcd{% endfor %}
+"""
+    return dind_cluster_config_j2
+
+
+def create_cluster(cluster_config_file):
+    # attempt to resolve unknown random rke up errors
+    for _ in range(3):
+        try:
+            return subprocess.check_output(
```

Comment:
Catch your error output by adding `stderr=subprocess.STDOUT` to the call to `check_output` then your error message will be there.